src/url.c

```diff
@@ -40,7 +41,10 @@ _parseHostAndPort(natsUrl *url, char *host, bool uInfo)
     if (sport != NULL)
     {
         if (sport[1] != '\0')
-            url->port = atoi(sport + 1);
+        {
+            if (!nats_ParseInt(sport + 1, &(url->port)))
```

Comment:
I would have no problem not supporting white spaces (for instance the Go client would reject any white space or non numerical value). So we could simply here invoke nats_ParseInt64 (would have first to determine the length of the remaining string), and since the port is a `int` then fail if result is negative or greater than INT32_MAX. We can use `return nats_setError(NATS_INVALID_ARG, "invalid port: '%s'", sport+1);` to be more informative about the error.

Reply:
I'll check better tomorrow (due to the timezone). BTW I started using `nats_ParseInt64()`, but if I'm not wrong the test-case `ConnectToWithMultipleURLs` has a sub-case with trailing spaces that `atoi()` instead manage, since it stops at the first non numerical value. So the length of the remaining string should not include spaces and, at the end, it is like using nats_ParseInt64() + discarding spaces + overflow check on int. This was the reason of `nats_ParseInt()`.